### PR TITLE
Blast refactor

### PR DESCRIPTION
**Pull request by cdsupina**
For #48.

I ended up moving `velocity_factor` from the Blast component to the AutoBlasterComponent as `velocity_multiplier`. This had the effect of neutralizing the influence of the former `velocity_factor` from blasts from the Player because there is no ManualBlasterComponent yet #49. If you run the game you will notice that blasts from the player are much more influenced by the velocity of the player because they aren't being scaled.

### CODE REVIEW

File: src/constants/mod.rs [hunk 1]
```diff
@@ -96,6 +96,10 @@ pub const DEFENSE_BAR_LIMIT: f32 = 63.0;
 pub const ROLL_BAR_X: f32 = 324.0;
 pub const ROLL_BAR_Y: f32 = 177.0;
 pub const ROLL_BAR_LIMIT: f32 = 28.0;
-pub const RESTOCK_BAR_X: f32 = 324.0; //should appear in arena
+pub const RESTOCK_BAR_X: f32 = 324.0;
 pub const RESTOCK_BAR_Y: f32 = 90.0;
 pub const RESTOCK_BAR_LIMIT: f32 = 28.0;
+
+// blasts
+pub const BLAST_MAX_SPEED_X: f32 = 1000.0; // blasts don't accelerate so this value should never be used
```
Comment (tigleym) [inline src/constants/mod.rs:104]: I think if we don't plan on using this value, we should remove it.

**Review by tigleym — COMMENTED** [review]
Thanks! This looks good to me. I think the only issues I had here was some lint errors this refactor introduced and also the usage of `BLAST_MAX_SPEED_X`.

File: src/constants/mod.rs (hunk 1 above)
Comment (cdsupina) [reply src/constants/mod.rs:104]: The value is just used for creating the motion2d component for the blast. I will just hard code the value for now. I could see us deciding to remove the max_speed property when we rework the motion2d logic.

**Comment by cdsupina** [comment]
I addressed your comments and also changed `Blast` to `BlastComponent`. If you are satisfied with the changes you can go ahead and merge it.

**Pull request merged**